torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -636,13 +625,57 @@ Tensor repeat_backward(Tensor grad, IntArrayRef repeats, IntArrayRef input_shape
   for (int64_t i = 0; i < num_unsqueezed; ++i) {
     grad = grad.sum(0, false);
   }
-  for (size_t j = num_unsqueezed; j < repeats.size(); ++j) {
+
+  const auto & grad_sizes = grad.sizes();
+  std::vector<int64_t> grad_sizes_(input_dims);
```

Comment:
We prefer DimVector to `std::vector<int64_t>` most of the time. https://github.com/pytorch/pytorch/blob/daf2a6a29da5109157bb44fc5e4dbc4fb5c56c15/aten/src/ATen/core/DimVector.h#L11.

The first 5 elements of DimVector go on the stack and the rest go onto the heap. c.f. all elements of std::vector go on the heap. DimVector improves performance for keeping track of metadata that has 5 or fewer elements, which is the case for most PyTorch use cases out in the wild